# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IndexedDatabaseManager.h
# Commit: 7e9d56eb485e
# Full Hash: 7e9d56eb485e06cca2a82fe88e2ae36c2ccd2095
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-04-01 21:26:59
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileManager. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65549
# ==============================================================================

diff -r a753c018a4a8 -r 7e9d56eb485e dom/indexedDB/IndexedDatabaseManager.h
--- a/dom/indexedDB/IndexedDatabaseManager.h	Wed Apr 01 09:54:24 2020 +0000
+++ b/dom/indexedDB/IndexedDatabaseManager.h	Wed Apr 01 09:52:54 2020 +0000
@@ -13,6 +13,7 @@
 #include "mozilla/Mutex.h"
 #include "nsClassHashtable.h"
 #include "nsHashKeys.h"
+#include "SafeRefPtr.h"
 
 namespace mozilla {
 
@@ -103,11 +104,11 @@
 
   void ClearBackgroundActor();
 
-  MOZ_MUST_USE RefPtr<FileManager> GetFileManager(
+  MOZ_MUST_USE SafeRefPtr<FileManager> GetFileManager(
       PersistenceType aPersistenceType, const nsACString& aOrigin,
       const nsAString& aDatabaseName);
 
-  void AddFileManager(FileManager* aFileManager);
+  void AddFileManager(SafeRefPtr<FileManager> aFileManager);
 
   void InvalidateAllFileManagers();
 